implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -266,19 +187,262 @@ impl Migrator {
 
     /// Run all migrations
     pub async fn migrate(&self, pool: &SqlitePool) -> Result<()> {
-        self.migrate_partial(pool, 0, i64::MAX, false).await
+        self.migrate_partial(pool, VersionInterval::from_to(0, i64::MAX))
+            .await
     }
 }
 
 #[cfg(test)]
 impl Migrator {
     /// Migrate the schema of the database right before the specified version
-    pub(crate) async fn migrate_before(
+    pub(crate) async fn migrate_interval(
         &self,
         pool: &SqlitePool,
-        version: i64, // not including
-        run_last_sql: bool,
+        version_interval: VersionInterval,
+    ) -> Result<()> {
+        self.migrate_partial(pool, version_interval).await
+    }
+}
+
+trait Migration {
+    fn version(&self) -> Version;
+}
+
+type Version = i64;
+
+pub(crate) struct VersionInterval {
```

Comment:
That's a nice improvement, but at this point I wonder if we could simply use `core::ops::Range`?

Reply:
Oh, I didn't have this in mind, I think using that would simplify it, indeed.

Reply:
I've removed this completely. Now that we store what migrations have been migrated and skip them, we were always using `0` as the lower bound of the interval, which makes using a range unnecessary.